common/lib/capa/capa/templates/jsinput.html

```diff
@@ -1,11 +1,12 @@
+<%page expression_filter="h"/>
 <%! from openedx.core.djangolib.markup import HTML %>
-<section id="inputtype_${id}" class="jsinput"
+<div id="inputtype_${id}" class="jsinput"
     data="${gradefn}"
     % if saved_state:
-    data-stored="${saved_state|x}"
+    data-stored="${saved_state}"
```

Comment:
Given that we are writing HTML, I think HTML-escaping is accurate.  I have been wrong before.  I do not know why someone would have chosen XML escaping to begin with.  Note that even if saved_state were JavaScript, you would want HTML-escaping here, and not JavaScript-escaping, because you are writing to HTML.